__tests__/Auth0Client.helpers.ts

```diff
@@ -61,7 +61,13 @@ export const loginWithRedirectFn = (mockWindow, mockFetch, fetchResponse) => {
   ) => {
     await auth0.loginWithRedirect(options);
     expect(mockWindow.location.assign).toHaveBeenCalled();
-    window.history.pushState({}, '', `/?code=${code}&state=${state}`);
+
+    if (code || state) {
+      window.history.pushState({}, '', `/?code=${code}&state=${state}`);
+    } else {
+      window.history.pushState({}, '', `/`);
+    }
+
```

Comment:
This allows to make the redirect fail so that we can react to unsuccesful calls to /authorize

Reply:
Do we want to cover the error scenario here? Auth0 will return `state` but also either `code` OR `error` depending on what happened. Do we have a test that covers that right now, or is it in the pipeline to be refactored?

Reply:
I added a test that relies on not having a code. But I can change it to have an error instead and update the tests accordingly

Reply:
Updated the PR. I will rework the function signature of the loginWithRedirect and loginWithPopup in a follow up PR as mentioned.